### ISSUE TITLE
[Bug] `limit.historyCount.suggestContinueAsNew=50` dynamic dev server config not working on latest RC version

### ISSUE THREAD

**Issue by cretz** [temporalio/cli#532]
### Describe the bug

Running `temporal server start-dev --dynamic-config-value limit.historyCount.suggestContinueAsNew=50` gives this warning:

```
level=WARN msg="Failed to convert value, using default" key=limit.historyCount.suggestContinueAsNew ignored-value=50 error="value type is not int"
```

Need this type of fix from older CLI: https://github.com/temporalio/cli/blob/7267df668277a9b43c50d454af50ff5e2b9f8e90/server/commands.go#L379-L380

**Comment by dnr** [comment]
It'd be a totally reasonable request to fix that on the server side instead

**Comment by cretz** [comment]
:+1: I'd accept that if y'all want to do it. Still will convert to int on this side because it's also reasonable for the server side to accept strict types and the (programmatic) caller to make sure they conform. I just never had a test for a integer-based dyn config here.

**Comment by cretz** [comment]
Yes, thanks! This is no longer an issue in latest CLI (we also use this specific dynamic config value in some SDK test suites, so we can confirm it works). Closing issue.
